wagtail/admin/views/generic/models.py

```diff
@@ -186,26 +234,51 @@ def search_queryset(self, queryset):
         }
         return queryset.filter(**filters)
 
+    def _get_title_column(self, column_class=TitleColumn):
+        def title_accessor(obj):
+            draftstate_enabled = self.model and issubclass(self.model, DraftStateMixin)
+
+            if draftstate_enabled and obj.latest_revision:
+                return obj.latest_revision.object_str
+            return str(obj)
+
+        return column_class(
+            "name",
+            label=gettext_lazy("Name"),
+            accessor=title_accessor,
+            get_url=self.get_edit_url,
+        )
+
+    def _get_updated_at_column(self, column_class=DateColumn):
+        return column_class("_updated_at", label=_("Updated"), sort_key="_updated_at")
```

Comment:
Issue: I'm unable to reverse the ordering of the "Updated" column on snippets. Clicking it once adds ``?ordering=_updated_at`` and sorts the listing by updated, but clicking it again does not change it to ``?ordering=-_updated_at``.

Reply:
Oops, sorry! Made a mistake when handling null values. Should be fixed now and I've added the tests to ensure it. If you reviewed each commit, that means you just need to re-review the two last commits. Thanks!